<launch>
    <arg name="device_id" default="0" />
    <arg name="frame_rate" default="30" />
    <arg name="image_width" default="640" />
    <arg name="image_height" default="480" />
    
    <!-- Launch RealSense cameras for chassis, hand, and top positions -->
    <include file="$(find camera_driver)/launch/realsense_cameras.launch" />
    
    <!-- Launch camera driver node for additional camera processing -->
    <node name="camera_driver_node" pkg="camera_driver" type="camera_driver_node.py" output="screen">
        <param name="device_id" value="$(arg device_id)" />
        <param name="frame_rate" value="$(arg frame_rate)" />
        <param name="image_width" value="$(arg image_width)" />
        <param name="image_height" value="$(arg image_height)" />
    </node>
</launch>